app/utils/write_rego.py

```diff
@@ -23,14 +22,27 @@ def write_to_file(rule: RequestObject) -> dict:
     # Initialize repository
     github.initialize()
 
-    result = initiate_rule + build_rego(rule.rules)
-
-    if result:
-        with open(file_path, "w") as file:
-            file.write(result)
-        # Update GitHub
-        github.push()
-        return {"status": "success", "message": "Policy successfully written to file"}
+    rule = {key: value for key, value in rule.items()}
+
+    # read the file
+    with open(file_path, "r") as file:
+        data = file.read() if operation == "write" else rule["old_state"]
+        result = (
+            data + build_rego(rule["rules"])
+            if data
+            else initiate_rule + build_rego(rule["rules"])
+        )
+
```

Comment:
We should be checking if the file exists first before attempting to open it. This breaks the POST route if the repo is empty.